### PR TITLE
[jax] Rewrite Qwen3 with abstraction linear layers

### PR DESCRIPTION
**Pull request by lk-chen**
# Description

Linear/Einsum layers added in #1482 . This PR rewrites Qwen models with added abstract layers.

(See whole plan in #1485 )

# Tests

CI, also manually:

```
# serve
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507
MODEL_IMPL_TYPE=flax_nnx vllm serve $CURRENT_VLLM_MODEL \
    --max-model-len=8192 \
    --disable-log-requests \
    --no-enable-prefix-caching

# benchmark
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507
vllm bench serve \
    --model $CURRENT_VLLM_MODEL \
    --num-prompts 1000 \
    --dataset-name random \
    --random-input-len 1024 \
    --random-output-len 4096

# eval
export CURRENT_VLLM_MODEL=Qwen/Qwen3-4B-Instruct-2507
lm_eval \
    --model local-completions \
    --model_args base_url=http://localhost:8000/v1/completions,model=$CURRENT_VLLM_MODEL \
    --tasks gsm8k_cot \
    --batch_size auto \
    --apply_chat_template \
    --num_fewshot 8
```

before
```
============ Serving Benchmark Result ============
Successful requests:                     1000      
Failed requests:                         0         
Benchmark duration (s):                  1681.23   
Total input tokens:                      1024000   
Total generated tokens:                  4096000   
Request throughput (req/s):              0.59      
Output token throughput (tok/s):         2436.32   
Peak output token throughput (tok/s):    3584.00   
Peak concurrent requests:                1000.00   
Total token throughput (tok/s):          3045.40   
---------------Time to First Token----------------
Mean TTFT (ms):                          617448.72 
Median TTFT (ms):                        630888.04 
P99 TTFT (ms):                           1347043.16
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          86.77     
Median TPOT (ms):                        75.31     
P99 TPOT (ms):                           127.02    
---------------Inter-token Latency----------------
Mean ITL (ms):                           86.77     
Median ITL (ms):                         64.86     
P99 ITL (ms):                            123.98    
==================================================
```
|  Tasks  |Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|---------|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k_cot|      3|flexible-extract|     8|exact_match|↑  |0.7582|±  |0.0118|
|         |       |strict-match    |     8|exact_match|↑  |0.0629|±  |0.0067|

after
```
============ Serving Benchmark Result ============
Successful requests:                     1000      
Failed requests:                         0         
Benchmark duration (s):                  1681.89   
Total input tokens:                      1024000   
Total generated tokens:                  4096000   
Request throughput (req/s):              0.59      
Output token throughput (tok/s):         2435.36   
Peak output token throughput (tok/s):    3584.00   
Peak concurrent requests:                1000.00   
Total token throughput (tok/s):          3044.19                                             <-------------- - 0.04%
---------------Time to First Token----------------
Mean TTFT (ms):                          617399.42 
Median TTFT (ms):                        630742.37 
P99 TTFT (ms):                           1347317.37
-----Time per Output Token (excl. 1st token)------
Mean TPOT (ms):                          86.78     
Median TPOT (ms):                        75.19     
P99 TPOT (ms):                           127.20    
---------------Inter-token Latency----------------
Mean ITL (ms):                           86.78     
Median ITL (ms):                         64.84     
P99 ITL (ms):                            123.87    
==================================================
```
|  Tasks  |Version|     Filter     |n-shot|  Metric   |   |Value |   |Stderr|
|---------|------:|----------------|-----:|-----------|---|-----:|---|-----:|
|gsm8k_cot|      3|flexible-extract|     8|exact_match|↑  |0.7582|±  |0.0118|
|         |       |strict-match    |     8|exact_match|↑  |0.0629|±  |0.0067|

### CODE REVIEW

File: tpu_inference/models/jax/qwen2_5_vl.py [hunk 1]
```diff
@@ -1142,7 +1131,12 @@ def load_weights(self, rng_key: jax.Array) -> None:
             })
 
         loader = self.WeightLoader(self.vllm_config, self.mesh)
-        loader.load_weights(self, mappings)
+        loader.load_weights(self,
```
Comment (jrplatin) [inline tpu_inference/models/jax/qwen2_5_vl.py:1134]: We are going to refactor weight loading soon to delegate to the layers, right?

**Comment by jrplatin** [comment]
Do you have E2E perf / accuracy testing showing that this doesn't introduce a regression?

**Comment by jrplatin** [comment]
Also, do the Qwen2/Qwen3 unit tests need to be updated?

File: tpu_inference/models/jax/qwen2_5_vl.py (hunk 1 above)
Comment (lk-chen) [reply tpu_inference/models/jax/qwen2_5_vl.py:1134]: Right, after the mapping on Line 1135 becomes empty. This way I can avoid layers within a model having different loading logic.

**Comment by lk-chen** [comment]
Updated unit test (CI passed)

and updated description to include benchmark/eval results.

**Review by jrplatin — APPROVED** [review]
